### PR TITLE
Fix SGCraft TileEntity Handling

### PR DESCRIPTION
**Pull request by draknyte1**
Version Bump.
Minor formatting changes.

### CODE REVIEW

File: src/base/gcewing/sg/BaseBlock.java [hunk 1]
```diff
@@ -6,609 +6,624 @@
...
+	}
+
+	public TileEntity getTileEntity(IBlockAccess world, BlockPos pos) {
+		TileEntity te = world.getTileEntity(pos.x, pos.y, pos.z); 
+		return te;
+	}
+	
+	public SGBaseTE getSGBaseTE(IBlockAccess world, BlockPos pos) {		
+		TileEntity te = getTileEntity(world, pos); 
+		if (SGBaseTE.class.isInstance(te)) {
```
Comment (Glease) [inline src/base/gcewing/sg/BaseBlock.java]: Why reflection?

File: src/mod/gcewing/sg/DHDBlock.java [hunk 2]
```diff
@@ -81,13 +77,13 @@ public boolean canHarvestBlock(IBlockState state, EntityPlayer player) {
...
     public void breakBlock(World world, BlockPos pos, IBlockState state) {
-        DHDTE cte = getTileEntity(world, pos);
+        TileEntity cte = getTileEntity(world, pos);
         super.breakBlock(world, pos, state);
         if (cte == null) {
             System.out.printf("DHDBlock.breakBlock: No tile entity at %s\n", pos);
         }
-        else if (cte.isLinkedToStargate) {
-            SGBaseTE gte = cte.getLinkedStargateTE();
+        else if (DHDTE.class.isInstance(cte) && ((DHDTE) cte).isLinkedToStargate) {
```
Comment (Glease) [inline src/mod/gcewing/sg/DHDBlock.java]: Again, why reflection?

File: src/mod/gcewing/sg/DHDBlock.java [hunk 3]
```diff
@@ -104,11 +100,13 @@ public boolean onBlockActivated(World world, BlockPos pos, IBlockState state, En
     
     public void checkForLink(World world, BlockPos pos) {
         //System.out.printf("DHDBlock.checkForLink at %s\n", pos);
-        DHDTE te = getTileEntity(world, pos);
-        if (te != null)
-            te.checkForLink();
-        else
-            System.out.printf("DHDBlock.breakBlock: No tile entity at %d\n", pos);
+        TileEntity te = getTileEntity(world, pos);
+        if (DHDTE.class.isInstance(te)) {
```
Comment (Glease) [inline src/mod/gcewing/sg/DHDBlock.java]: Same here.

**Review by Glease — COMMENTED** [review]
I can see a lot other instances where plain old instanceof can be used instead of reflection. They are semantically the same and should be on the same magnitudes of performance, but why?

**Comment by Dream-Master** [comment]
@draknyte1

**Comment by Dream-Master** [comment]
@Glease

**Review by Glease — APPROVED** [review]

**Pull request merged**